From e4b0fb3d817d9569cb915aeea793b92ccb798048 Mon Sep 17 00:00:00 2001
From: Avinash Kumar <avikumar@habana.ai>
Date: Wed, 10 Nov 2021 02:08:56 +0200
Subject: [PATCH] [SW-62043] greco: mlong calls for code relocation

zephyr mgmt some drivers are relocated from iccm to
sram. For this compiler needs to have mlong call option.

Signed-off-by: Avinash Kumar <avikumar@habana.ai>
---

diff --git a/package/zephyr/zephyr.mk b/package/zephyr/zephyr.mk
index fad44db..f9d794f 100644
--- a/package/zephyr/zephyr.mk
+++ b/package/zephyr/zephyr.mk
@@ -49,6 +49,9 @@
 ZEPHYR_CONF_OPTS += -DEXTRA_CFLAGS="-Werror -include $(HABANA_SOFTWARE_STACK)/embedded/specs/$(DEVICE_NAME)/$(DEVICE_NAME)_boot_info.h"
 ZEPHYR_CONF_OPTS_MGMT += -DBOARD=hl_$(DEVICE_NAME)_mgmt_$(HL_TARGET)
 ZEPHYR_CONF_OPTS_MGMT += -DDEVICE=$(DEVICE_NAME)
+ifeq ($(DEVICE_NAME),greco)
+ZEPHYR_CONF_OPTS_MGMT += -DEXTRA_CFLAGS=-mlong-calls
+endif
 ZEPHYR_CONF_OPTS_MGMT += $(dir $(BR2_EXTERNAL_HABANALABS_PATH))/arcmgmt
 ZEPHYR_CONF_OPTS_MGMT += -DZEPHYR_MODULES=$(dir $(BR2_EXTERNAL_HABANALABS_PATH))/zephyr-modules/eeprom
 ifeq ($(HL_TARGET),pldm)
